### PR TITLE
Same timer and on-demand popup for all OS

### LINKED ISSUE #1520: Parameter editor is slow to build when there are many parameters in the current step
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Description of the bug

The parameter editor window is slow when entering a sequence step that has many parameters

### Steps To Reproduce

When entering step 12 is a good example of the slow down

### Expected behavior

It should always be fast, irrespective of the number of parameters to display.

I guess the problem is that on windows and linux all tooltips get created when entering a new sequence step, instead of only building them on demand as macOS does.

### Screenshots / Logs

_No response_

### Software information

- Operating system: Windows
- arduPilot Methodic Configuration version: 2.11.1

### Additional context

@OmkarSarkar204 Does macOS slow down on step 12? can that solution be ported to the other two OS?

### PR DESCRIPTION
**Pull request by OmkarSarkar204**
# Description

Fix: #1520

Parameter ediotr builder is slow on windows and linux, applying the same logic for popups as on MacOS to the other OS's

## Checklist

- [X] Run pre-commit checks locally
- [X] Verified by a human programmer
- [X] All commits are signed off (use `git commit --signoff`)
- [X] Code follows our [coding standards](../CONTRIBUTING.md#code-quality--standards)
- [X] Documentation updated if needed
- [X] No breaking changes or properly documented

## Testing

Describe how you tested these changes:

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing performed
- [ ] Tested on flight controller hardware

## WIP
I want to test this on all the three OS's to see any edge cases.

### CODE REVIEW

File: ardupilot_methodic_configurator/frontend_tkinter_show.py [hunk 1]
```diff
@@ -517,14 +497,14 @@ def _cancel_timer(self, name: str) -> None:
...
-            self.tooltip.deiconify()
-            Tooltip._active_tooltip = self
+    # def show(self, event: Optional[tk.Event] = None) -> None:  # pylint: disable=unused-argument
+    #     """On non-macOS, tooltip already exists, show it on events."""
+    #     self._cancel_hide()
+    #     self._hide_active_tooltip()
+    #     if self.tooltip:
+    #         self.position_tooltip()
+    #         self.tooltip.deiconify()
+    #         Tooltip._active_tooltip = self
```
[the remark this answers is not in the document]
Comment (OmkarSarkar204) [reply ardupilot_methodic_configurator/frontend_tkinter_show.py:508]: WIP

File: ardupilot_methodic_configurator/frontend_tkinter_show.py [hunk 2]
```diff
@@ -652,35 +632,32 @@ def position_tooltip(self) -> None:
...
         self._cancel_hide()
         self._cancel_timer("alpha")
         if self.tooltip:
-            if platform_system() == "Darwin":
-                self.tooltip.destroy()
-                self.tooltip = None
-            else:
-                self.tooltip.withdraw()
+            self.tooltip.destroy()
+            self.tooltip = None
```
[the remark this answers is not in the document]
Comment (OmkarSarkar204) [reply ardupilot_methodic_configurator/frontend_tkinter_show.py:655]: We can hide them instead of destroying everytime.

[the remark this answers is not in the document]
Comment (OmkarSarkar204) [reply ardupilot_methodic_configurator/frontend_tkinter_show.py:633]: I looked into hiding the tooltips instead of destroying them. Unfortunately, Tkinter on macOS has a bug where calling `.withdraw()` and then `.deiconify()` on a borderless window permanently disturbs the MacWindowStyle, resulting in a black box.
Furthermore, while creating 800 Toplevels at once causes a freeze, creating a single Toplevel on-demand after a 250ms hover delay takes <1ms and causes zero UI flicker on Windows. To keep it unified without OS checks, I think sticking to destroying them on Leave globally is a better option.

**Comment by OmkarSarkar204** [comment]
@amilcarlucas , 
I found that the tooltips were not the main problem of the slow load. The true issue is positively the sheer volume of parameters being loaded into the Tkinter UI at the exact same time. The data itself loads very fast, but the Tkinter rendering completely chokes when trying to instantiate and .grid() hundreds of complex row widgets (Comboboxes, Entries, Labels) simultaneously.

I used a scroll method to load a small amount of parameters. The scroll is currently the most efficient method i have explored and tested to make it fast till now 

Though i think its a bypass and will have a few bugs itself, but works perfectly for the time being, will test other methods also but it is much faster than the original way.

Working perfectly on Windows and Mac, Linux testing and updating tests are still left , ill update it shortly

Please let me know if anything can be improved.

**Comment by OmkarSarkar204** [comment]
@amilcarlucas , any changes needed in this?

**Comment by amilcarlucas** [comment]
Thanks, it looks good, I will need some time to review and I want to benchmark it a bit.

In the future I plan to use the new [python 3.15 unintrusive profiler](https://docs.python.org/3.15/whatsnew/3.15.html#whatsnew315-sampling-profiler) to find bottlenecks without requiring code instrumentation.

**Comment by amilcarlucas** [comment]
Using copilot in VS code you can add this github pull request as reference and tell it to validate the review comments and address the valid ones.

**Comment by OmkarSarkar204** [comment]
Thanks, I have never used copilot in VS but ill try it

**Comment by amilcarlucas** [comment]
I rebased it, force pushed it and tested it. It works great!!!!
Good work

**Comment by OmkarSarkar204** [comment]
Thank You so much.

**Comment by OmkarSarkar204** [comment]
Adressed copilot review.
Tested Mac, Linux

**Comment by amilcarlucas** [comment]
I simplified this massively in #1572 and added the rest to #1573. I did an extra second commit on #1573 on top of your original changes

**Comment by OmkarSarkar204** [comment]
@amilcarlucas , Glad this worked well and also thank you very much for the appreciation on discord, is it ok to close this one.

**Comment by amilcarlucas** [comment]
Yes, we'll continue this on #1573 

I can only test it on linux and windows. Can you test it on macOS?

**Comment by OmkarSarkar204** [comment]
> Yes, we'll continue this on #1573
> 
> I can only test it on linux and windows. Can you test it on macOS?

Sure ill do it and let you know :)
